src/BaseComponent.jsx

```diff
@@ -178,7 +179,9 @@ export default class Component extends React.PureComponent {
                 if (prefixIndex === 0) {
                     childName = childName.substring(prefix.length);
                 }
-                childAttributes[childName] = this.props[name];
+                if (!prefix || prefixIndex === 0) {
+                    childAttributes[childName] = this.props[name];
```

Comment:
At some point we can rename this variable to `undeclaredAttributes` :P (doesn't have to be now)